datafusion/optimizer/src/analyzer/type_coercion.rs

```diff
@@ -902,19 +902,17 @@ mod test {
     #[test]
     fn agg_udaf_invalid_input() -> Result<()> {
         let empty = empty();
-        let return_type: ReturnTypeFunction =
-            Arc::new(move |_| Ok(Arc::new(DataType::Float64)));
-        let state_type: StateTypeFunction =
-            Arc::new(move |_| Ok(Arc::new(vec![DataType::UInt64, DataType::Float64])));
+        let return_type = DataType::Float64;
+        let state_type = vec![DataType::UInt64, DataType::Float64];
         let accumulator: AccumulatorFactoryFunction =
             Arc::new(|_| Ok(Box::<AvgAccumulator>::default()));
-        let my_avg = AggregateUDF::new(
-            "MY_AVG",
-            &Signature::uniform(1, vec![DataType::Float64], Volatility::Immutable),
-            &return_type,
-            &accumulator,
-            &state_type,
-        );
+        let my_avg = AggregateUDF::from(SimpleAggregateUDF::new_with_signature(
+            "MY_AVG".to_string(),
```

Comment:
Since `new_with_signature` takes `impl Into<String>` I think this would also work:

```suggestion
            "MY_AVG",
```